File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 7/31/2025, 12:22:50 PM
class Solution {
    public boolean isPalindrome(int x) {
        int sum=0;
        for(int i=x;i>0;i=i/10){
            int rem=i%10;
            sum=sum*10+rem;
        }
        if(sum==x)
        return true;
        else
        return false;
        
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.py
# Last updated: 7/31/2025, 12:22:56 PM
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0:
           return False

        reversed_num=0
        temp=x

        while temp!=0:
            digit=temp%10
            reversed_num=reversed_num*10+digit
            temp//=10
        return reversed_num==x        